documentation/FortranForCProgrammers.md

```diff
@@ -0,0 +1,254 @@
+  generations of computer systems hardware and software.  Standard
+  compliance notwithstanding, Fortran programmers generally expect that
+  code that has compiled successfully in the past will continue to
+  compile and work indefinitely.  The standards sometimes designate
+  features as being deprecated, obsolescent, or even deleted, but that
+  can be read only as discouraging their use in new code -- they'll
+  probably always work in any serious implementation.
+* Fortran has two source forms, which are typically distinguished by
+  filename suffixes.  `foo.f` is old-style "fixed-form" source, and
+  `foo.f90` is new-style "free-form" source.  All language features
+  are available in both source forms.  Neither form has reserved words
+  in the sense that C does.  Spaces are not required between tokens
+  in fixed form, and case is not significant in either form.
+* Variable declarations are optional by default.  Variables whose
+  names begin with the letters `I` through `N` are implicitly
+  `INTEGER`, and others are implicitly `REAL`.  These implicit typing
+  rules can be changed in the source.
+* Fortran uses parentheses for both array references and for function calls.
+  All arrays must be declared as such; other names followed by parenthesized
+  expressions are assumed to be function calls.
+* Fortran has a _lot_ of built-in "intrinsic" functions.  They are always
+  available without a need to declare or import them.  Their names reflect
+  the implicit typing rules, so you will encounter names that have been
+  modified so that they have the right type (e.g., `AIMAG` has a leading `A`
+  so that it's `REAL` rather than `INTEGER`).
+* The modern language has means for declaring types, data, and subprogram
+  interfaces in compiled "modules", as well as legacy mechanisms for
+  sharing data and interconnecting subprograms.
+
+Data Types
+----------
+There are five built-in ("intrinsic") types: `INTEGER`, `REAL`, `COMPLEX`,
+`LOGICAL`, and `CHARACTER`.
+They are parameterized with "kind" values, which should be treated as
+non-portable integer codes but in practice today are the byte sizes of
+the data.
+The legacy `DOUBLE PRECISION` intrinsic type is an alias for a kind of `REAL`
+that should be bigger than the default `REAL`.
+
+`COMPLEX` is a simple structure that comprises two `REAL` components.
+
+`CHARACTER` data also have length, which may or may not be known at compilation
+time.
+`CHARACTER` variables are fixed-length strings and they get padded out
+with space characters when not completely assigned.
+
+User-defined ("derived") data types can be synthesized from the intrinsic
+types and from previously-defined user types, much like a C `struct`.
+Derived types can be parameterized with integer values that either have
+to be constant at compilation time ("kind" parameters) or deferred to
+execution ("len" parameters).
+
+Derived types can inherit ("extend") from one other derived type, no more.
```

Comment:
I read "no more" as in "this is not longer in the language".  Perhaps write "from exactly one other derived type" or something else.

Reply:
Rewording.